influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -253,145 +240,68 @@ pub struct BufferedWrite {
     pub response_tx: oneshot::Sender<BufferedWriteResult>,
 }
 
-#[derive(Debug, Default, Eq, PartialEq)]
+#[derive(Debug, Default)]
 pub struct BufferedData {
     database_buffers: HashMap<String, DatabaseBuffer>,
 }
 
-#[derive(Debug, Default, Eq, PartialEq)]
-struct DatabaseBuffer {
-    table_buffers: HashMap<String, TableBuffer>,
-}
-
-#[derive(Debug, Clone, Eq, PartialEq)]
-pub struct TableBuffer {
-    pub segment_key: PartitionKey,
-    rows: Vec<Row>,
-    timestamp_min: i64,
-    timestamp_max: i64,
-}
-
-impl TableBuffer {
-    pub fn add_rows(&mut self, rows: Vec<Row>) {
-        self.rows.reserve(rows.len());
-        for row in rows {
-            self.timestamp_min = self.timestamp_min.min(row.time);
-            self.timestamp_max = self.timestamp_max.max(row.time);
-            self.rows.push(row);
-        }
-    }
-
-    pub fn timestamp_min_max(&self) -> TimestampMinMax {
-        TimestampMinMax {
-            min: self.timestamp_min,
-            max: self.timestamp_max,
+impl BufferedData {
+    /// Returns the table data as record batches
+    pub(crate) fn table_record_batches(
+        &self,
+        db_name: &str,
+        table_name: &str,
+        schema: &Schema,
+    ) -> Option<Vec<RecordBatch>> {
+        if let Some(db_buffer) = self.database_buffers.get(db_name) {
+            if let Some(table_buffer) = db_buffer.table_buffers.get(table_name) {
+                return Some(table_buffer.record_batches(schema));
+            }
```

Comment:
you can condense this like so using [`and_then`](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then)
```suggestion
        self.database_buffers
            .get(db_name)
            .and_then(|db_buffer| db_buffer.table_buffers.get(table_name))
            .and_then(|table_buffer| Some(table_buffer.record_batches(schema))
```

This will let you also get rid of the `None` below